### PR TITLE
feat: rename execute_record_counts_smoke_test to run_connector_readiness_test_report

### PR DESCRIPTION
**Pull request by aaronsteers**
# feat: rename execute_record_counts_smoke_test to run_connector_readiness_test_report

## Summary

This PR renames the `execute_record_counts_smoke_test` function to `run_connector_readiness_test_report` and significantly enhances its behavior to generate a comprehensive markdown-formatted readiness report instead of returning a structured object. The changes address feedback that LLMs were using this function incorrectly due to its confusing name and improve its utility as a final readiness assessment tool.

**Key Changes:**
- **Function renamed** from `execute_record_counts_smoke_test` to `run_connector_readiness_test_report`
- **Return type changed** from `MultiStreamSmokeTest` object to markdown-formatted string
- **Added validation warnings** for suspicious record counts (multiples of 10, 1 record, 0 records)
- **Added field count validation** to ensure records have at least 2 fields
- **Enhanced reporting** with stream subset information ("X out of Y streams") and detailed per-stream statistics
- **Preserved error handling** behavior for failed streams

## Review & Testing Checklist for Human

- [ ] **Test with real connector manifests** - Verify the markdown output is readable and accurate with actual connectors (not just test data)
- [ ] **Validate field count warnings** - Test with connectors that have records with 1 field vs 2+ fields to ensure the validation logic works correctly
- [ ] **Check record count warnings** - Test edge cases like exactly 10 records, 1 record, 0 records to verify warnings trigger appropriately
- [ ] **Verify MCP integration** - Restart the MCP server and confirm the renamed function is properly registered and callable
- [ ] **Performance testing** - Test with larger datasets to ensure the new `include_records_data=True` calls don't cause significant slowdowns

**Recommended Test Plan:**
1. Test the function directly with various connector manifests (Rick and Morty, real connectors)
2. Verify the MCP server recognizes the new function name after restart
3. Check that the chatmode configuration correctly references the new function
4. Test edge cases: empty streams, single-field records, pagination scenarios

---

### Diagram

```mermaid
%%{ init : { "theme" : "default" }}%%
graph TD
    ValidationTesting["connector_builder_mcp/<br/>_validation_testing.py"]:::major-edit
    ConnectorBuilder["connector_builder_mcp/<br/>_connector_builder.py"]:::minor-edit
    ChatmodeConfig[".github/chatmodes/<br/>connector-builder-mcp-test-mode<br/>.chatmode.md"]:::minor-edit
    MCPServer["MCP Server<br/>Registration"]:::context
    
    
    ValidationTesting -->|"imports function"| ConnectorBuilder
    ConnectorBuilder -->|"registers as tool"| MCPServer
    ChatmodeConfig -->|"references tool name"| MCPServer
    
    ValidationTesting -->|"calls with include_records_data=True"| StreamTestRead["execute_stream_test_read<br/>(existing function)"]:::context
    ValidationTesting -->|"generates markdown report"| MarkdownOutput["Markdown Report<br/>with warnings"]:::context
    
    subgraph Legend
        L1[Major Edit]:::major-edit
        L2[Minor Edit]:::minor-edit  
        L3[Context/No Edit]:::context
    end
    
    classDef major-edit fill:#90EE90
    classDef minor-edit fill:#87CEEB
    classDef context fill:#FFFFFF
```

### Notes

- **Breaking Change**: This is a breaking API change as the return type completely changed from structured object to string
- **Performance Impact**: The function now calls `execute_stream_test_read` with `include_records_data=True` to enable field count validation, which may be slower for large datasets
- **TODO Added**: Included commented placeholder for future page size validation as requested
- **All Tests Pass**: Code quality checks and existing unit tests (76 tests) all pass successfully

**Session Info**: Requested by AJ Steers (@aaronsteers) - [Devin Session](https://app.devin.ai/sessions/9ba77fd4ec5b43b5921567c30c803729)

> [!IMPORTANT]
> **Auto-merge enabled.**
> 
> _This PR is set to merge automatically when all requirements are met._

### CODE REVIEW

File: connector_builder_mcp/_validation_testing.py [hunk 1]
```diff
@@ -436,7 +437,7 @@ def execute_record_counts_smoke_test(
                 config=config,
                 stream_name=stream_name,
                 max_records=max_records,
-                include_records_data=False,
+                include_records_data=True,
```
Comment (aaronsteers) [inline connector_builder_mcp/_validation_testing.py:440]: Let's keep `include_records_data=False` and add a new `include_record_stats` boolean. Our `include_record_stats` arg will (when selected) give basic stats on the record properties:

```
{  num_properties: n,
   properties:
     columnA: {
       type: str,
       num_null: 17,
       num_non_null: 14,
     }.
     columnB: {...},
     ...,
}
```

And only for top-level columns.

File: connector_builder_mcp/_validation_testing.py [hunk 2]
```diff
@@ -387,22 +387,23 @@ def execute_record_counts_smoke_test(
...
         Field(description="Optional path to .env file for secret hydration"),
     ] = None,
-) -> MultiStreamSmokeTest:
-    """Execute a smoke test to count records from all streams in the connector.
+) -> str:
+    """Execute a connector readiness test and generate a comprehensive markdown report.
 
     This function tests all available streams by reading records up to the specified limit
-    and returns comprehensive statistics including record counts, errors, and timing information.
+    and returns a markdown-formatted readiness report with validation warnings and statistics.
```
Comment (aaronsteers) [inline connector_builder_mcp/_validation_testing.py:394]: Make a note here that this test is meant to be run individual streams have been tested with the test read tool, to validate things are working properly, and to generate a report we can share back with the end user.

File: connector_builder_mcp/_validation_testing.py [hunk 3]
```diff
@@ -249,6 +251,10 @@ def execute_stream_test_read(
         bool,
         Field(description="Whether to include actual record data in the response"),
     ] = False,
+    include_record_stats: Annotated[
+        bool,
+        Field(description="Include basic statistics on record properties"),
+    ] = False,
```
Comment (aaronsteers) [inline connector_builder_mcp/_validation_testing.py:257]: Default to enabled, since this doesn't hit the context window as hard.

```suggestion
    ] = True,
```
